google-cloud-pubsub/lib/google/cloud/pubsub/service.rb

```diff
@@ -175,11 +175,21 @@ def list_subscriptions options = {}
         ##
         # Creates a subscription on a given topic for a given subscriber.
         def create_subscription topic, subscription_name, options = {}
-          push_config = if options[:endpoint]
+          push_config = if options[:push_config]
+                          # Handle if push config is provided when creating the subscription
+                          if options[:push_config].is_a? Google::Cloud::PubSub::Subscription::PushConfig
+                            options[:push_config].to_grpc
+                          elsif !options[:push_config].is_a? Google::Cloud::PubSub::V1::PushConfig
+                            raise ArgumentError "push_config is not a valid pubsub PushConfig object"
+                          else
+                            options[:push_config]
```

Comment:
Would this be anything other than a `Hash`? Might be safer to explicitly test for that, and otherwise raise `ArgumentError` in the final `else` block.

Reply:
Although in the end this becomes a hash, I thought it would be better to pass the objects instead, and reject everything else (including the hash).

This if clause is a bit confusing, because the raise is in the condition above.
I'll restructure the if condition into a more orthodox one raising only on the `else` block.

Reply:
Thanks for pointing this out. There was an actual bug on the raise branch of this `if` that was discovered when the tests were added :)